rpc/client.go

```diff
@@ -109,6 +135,84 @@ func (opts *UploadReportOptions) artifactConsumer(ctx context.Context, testChan
 	}
 }
 
+func getSignedURL(task string, execution int, awsRegion string, data []byte, awsSecretKey string, awsAccessKey string, resultsHandlerHost string, resultType string) (string, error) {
+	service := "execute-api"
+	if awsSecretKey == "" || awsAccessKey == "" || resultsHandlerHost == "" || resultType == "" {
+		return "", errors.New("Getting signed URL failed. AWS secret key, AWS access key, resuls handler host and result type required.")
+	}
+	client := &http.Client{}
+	name := uuid.New()
+	url := fmt.Sprintf("%s/evergreen/%s/%s/%s/%s", resultsHandlerHost, task, strconv.Itoa(execution), resultType, name.String())
```

Comment:
Maybe a comment linking to the 10gen/data-pipes repo where we should have docs for this API after the current PR in that repo goes in?